ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -151,7 +151,9 @@ internal constructor(
           responses.forEach { (task, taskData) ->
             taskDataDeltas.add(TaskDataDelta(task.id, task.type, Optional.ofNullable(taskData)))
           }
-          submissionRepository.createOrUpdateSubmission(it, taskDataDeltas.build(), isNew = true)
+          submissionRepository
```

Comment:
If the returned subscription is disposed of before the update completes, won't the update process be interrupted? Can we instead introduce an `externalScope` as defined [here](https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad), and call `submissionRepository.createOrUpdateSubmission(..).blockingFoo()` from a coroutine?

It's also not clear why this method returns a Single at all. Could this be simplified?

Reply:
Done